compiler/rustc_hir_analysis/src/impl_wf_check.rs

```diff
@@ -183,6 +214,9 @@ fn report_unused_parameter(
     span: Span,
     kind: &str,
     name: Symbol,
+    is_impl_trait: bool,
+    is_bound_to_projection: bool,
+    is_bound_to_circular_projection: bool,
```

Comment:
I haven't read the rest of your impl, this is just drive-by commentary.

Could please turn these `bool`s into custom enums? And merge these three flags as much as possible unless they are already mutually exclusive?

Reply:
Yep 👍 `is_impl_trait` is orthogonal to the other two booleans, but `is_bound_to_projection` and `is_bound_to_circular_projection` can be merged